recipes/libnvjitlink/meta.yaml

```diff
@@ -0,0 +1,151 @@
+{% set target_name = "sbsa-linux" %}  # [aarch64]
+{% set target_name = "x64" %}  # [win]
+{% set extension = "tar.xz" %}  # [not win]
+{% set extension = "zip" %}  # [win]
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://developer.download.nvidia.com/compute/cuda/redist/{{ name }}/{{ platform }}/{{ name }}-{{ platform }}-{{ version }}-archive.{{ extension }}
+  sha256: 92a70db1ee90f1816a40d4d2019e0a59c43eebb9f3eb1208abedcc00a13e48be  # [linux64]
+  sha256: dd317e1d052b4231ab95d1b472105640f617f7d250aacc5bc1c952aab93d7483  # [ppc64le]
+  sha256: 9f18090f6587e95be0ee424f07358eeb0e15c5ab3f8636870f68dd84a92ed2e6  # [aarch64]
+  sha256: cae74b6790174b283d892bb20e1ff91bff13c8a2a8486de4f417e09675469372  # [win]
+
+build:
+  number: 0
+  skip: true  # [osx]
+
+outputs:
+  - name: libnvjitlink
+    files:
+      - lib/libnvJitLink.so.*                            # [linux]
+      - targets/{{ target_name }}/lib/libnvJitLink.so.*  # [linux]
+      - Library\bin\nvJitLink*.dll                       # [win]
+    requirements:
+      build:
+        - {{ compiler("c") }}
+        - {{ compiler("cxx") }}
+        - arm-variant * {{ arm_variant_type }}  # [aarch64]
+        - sysroot_{{ target_platform }} 2.17    # [linux]
+      host:
+        - cuda-version {{ cuda_version }}
+      run:
+        - {{ pin_compatible("cuda-version", max_pin="x.x") }}
+      run_constrained:
+        - arm-variant * {{ arm_variant_type }}  # [aarch64]
+    test:
+      commands:
+        - test -L $PREFIX/lib/libnvJitLink.so.{{ version }}                                          # [linux]
+        - test -L $PREFIX/lib/libnvJitLink.so.{{ version.split(".")[0] }}                            # [linux]
+        - test -L $PREFIX/targets/{{ target_name }}/lib/libnvJitLink.so.{{ version }}                # [linux]
+        - test -f $PREFIX/targets/{{ target_name }}/lib/libnvJitLink.so.{{ version.split(".")[0] }}  # [linux]
+        - if not exist %LIBRARY_BIN%\nvJitLink*.dll exit 1                                           # [win]
+    about:
+      home: https://developer.nvidia.com/cuda-toolkit
+      license_file: LICENSE
+      license: LicenseRef-NVIDIA-End-User-License-Agreement
+      license_url: https://docs.nvidia.com/cuda/eula/index.html
+      summary: CUDA nvJitLink library
+      description: |
+        nvJitLink - Just-in-Time Link Time Optimization (JIT LTO)
+      doc_url: https://docs.nvidia.com/cuda/nvjitlink/
+
+  - name: libnvjitlink-dev
+    build:
+      run_exports:
+        - {{ pin_subpackage("libnvjitlink", max_pin="x") }}
+    files:
+      - lib/libnvJitLink*.so                            # [linux]
+      - lib/pkgconfig                                   # [linux]
+      - targets/{{ target_name }}/include/nvJitLink.h   # [linux]
+      - targets/{{ target_name }}/lib/libnvJitLink*.so  # [linux]
+      #- targets/{{ target_name }}/lib/stubs             # [linux]  # bug: currently missing
```

Comment:
Is there an nvbug filed upstream? Have we confirmed that nvjitlink is _supposed_ to ship stubs?

Reply:
I don't know if a bug is filed (I bet there's), but this is documented:
> For a cross-compile toolkit (such as linux64 host, aarch64 target), we are missing the host-side stub library for libnvJitLink. As a workaround, you can copy the libnvJitlink.so from the target install (for example, /usr/local/cuda-12.1/targets/aarch64-linux/lib/libnvJitLink.so) to the host install (/usr/local/cuda-12.1/targets/aarch64-linux/lib/stubs/libnvJitLink.so). Similarly if you are using the static library version (/usr/local/cuda-12.1/targets/aarch64-linux/lib/libnvJitLink_static.a), can copy it from the target install (i.e. install on the device) to the same path on the host install. For a sbsa cross-compile replace aarch64 with sbsa in the above copies.

https://docs.nvidia.com/cuda/cuda-toolkit-release-notes/index.html#general-cuda-known-issues

Reply:
I see libnvJitLink stubs included in the release after 12.1.

Reply:
Awesome, thanks Alex. Let's keep this line then, and uncomment it when 12.2 is out. Mark as resolved.